lib/hooks.js

```diff
@@ -408,6 +409,52 @@ function hookIterator (fn, request, reply, next) {
   return fn(request, reply, next)
 }
 
+// wrapper that we expose to the user for hooks handling
+function addHook (name, fn) {
+  if (fn == null) {
+    throw new FST_ERR_HOOK_INVALID_HANDLER(name, fn)
+  }
+
+  if (name === 'onSend' || name === 'preSerialization' || name === 'onError' || name === 'preParsing') {
```

Comment:
instead of doing all these `||`, wouldn't be better to create a single set and check against it?

Reply:
Which one has the smallest memory footprint?